File: leetcode-04/solutions/python3/0057.py
class Solution:
    def insert(self, intervals: List[List[int]], newInterval: List[int]) -> List[List[int]]:
        ans = []
        index = len(intervals)

        for i in range(len(intervals)):
            if newInterval[0] < intervals[i][0]:
                index = i
                break
        intervals.insert(index, newInterval)

        for interval in intervals:
            if not ans or interval[0] > ans[-1][1]:
                ans.append(interval)
            else:
                ans[-1][1] = max(ans[-1][1], interval[1])

        return ans

File: leetcode-04/solutions/cpp/0057.cpp
class Solution {
 public:
  vector<vector<int>> insert(vector<vector<int>>& intervals,
                             vector<int>& newInterval) {
    vector<vector<int>> ans;
    auto it = intervals.begin();

    while (it != intervals.end() && (*it)[0] < newInterval[0]) it++;
    intervals.insert(it, newInterval);

    for (vector<int>& interval : intervals) {
      if (ans.empty() || interval[0] > ans.back()[1])
        ans.push_back(interval);
      else
        ans.back()[1] = max(ans.back()[1], interval[1]);
    }

    return ans;
  }
};